### ISSUE TITLE
ReactComponentB.domType specifiable too late

### ISSUE THREAD

**Issue by japgolly** [japgolly/scalajs-react#76]
This should, but doesn't, work.
```
 66     .domType[HTMLCanvasElement]
 67     .componentDidMount(scope => {
 69       val ctx = scope.getDOMNode().getContext("2d")
```

**Comment by japgolly** [comment]
This is actually a nightmare of types to fix. Would need to add a node type to ComponentScopeX
```scala
def componentDidMount(f: ComponentScopeM[P, S, B] => Unit)
```

Signatures are already verbose enough. Maybe should use dependent types... Either way, too large a fix for 0.8.1.

**Comment by japgolly** [comment]
Closed via 0178303d2f8decc61ee05de69914c3f5b1f36d41
